### ISSUE TITLE
`softplus` outputs `inf` for large inputs after converting to lite model

### ISSUE THREAD

**Issue by YaoJiayi** [tensorflow/tensorflow#60892]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source): 2.14.0-dev20230602

### 2. Code
This is the minimized code to reproduce the issue:
```python
import tensorflow as tf
import numpy as np

x1 = tf.constant([100., 100.], shape=[1, 2])

class Model(tf.keras.Model):

  def __init__(self):
    super(Model, self).__init__()

  @tf.function(input_signature=[tf.TensorSpec(x1.shape, x1.dtype)])
  def call(self, x):
    return tf.math.softplus(x)

# Initializing the model
m = Model()

m(x1)
print('Keras mode output: ', m(x1))

converter = tf.lite.TFLiteConverter.from_keras_model(m)
tflite_model = converter.convert()

def _evaluateTFLiteModel(tflite_model, input_data):
    interpreter = tf.lite.Interpreter(model_content=tflite_model)
    interpreter.allocate_tensors()

    input_details = interpreter.get_input_details()
    output_details = interpreter.get_output_details()

    for i in range(len(input_data)):
        interpreter.set_tensor(input_details[i]['index'], input_data[i])

    interpreter.invoke()
    output_data = [interpreter.get_tensor(output_details[i]['index'])
                   for i in range(len(output_details))]
    return output_data

print('Lite mode output: ', _evaluateTFLiteModel(tflite_model,[x1])[0])
```
### 3. Failure after conversion
Output:
```
Keras mode output:  tf.Tensor([[100. 100.]], shape=(1, 2), dtype=float32)
Lite mode output:  [[inf inf]]
```
Model produces wrong results:
- The original model will produce `[[100, 100]]`, which is the same as the input tensor `x1`
- After converting to tflite, the model may compute `softplus(x) = log(exp(x) + 1)` by first computing `exp`, which causes overflow `[[inf, inf]]`

**Comment by pkgoogle** [comment]
Just adding info:

Tensorflow performs this approximation/optimization here:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/softplus_op.h#32
```c++
struct Softplus {
  // Computes Softplus activation.
  //
  // features: any shape.
  // activations: same shape as "features".
  void operator()(const Device& d, typename TTypes<T>::ConstTensor features,
                  typename TTypes<T>::Tensor activations) {
    // Choose a threshold on x below which exp(x) may underflow
    // when added to 1, but for which exp(x) is always within epsilon of the
    // true softplus(x).  Offset of 2 from machine epsilon checked
    // experimentally for float16, float32, float64.  Checked against
    // softplus implemented with numpy's log1p and numpy's logaddexp.
    static const T threshold =
        Eigen::numext::log(Eigen::NumTraits<T>::epsilon()) + T(2);
    // Value above which exp(x) may overflow, but softplus(x) == x
    // is within machine epsilon.
    auto too_large = features > features.constant(-threshold);
    // Value below which exp(x) may underflow, but softplus(x) == exp(x)
    // is within machine epsilon.
    auto too_small = features < features.constant(threshold);
    auto features_exp = features.exp();
    activations.device(d) = too_large.select(
        features,                       // softplus(x) ~= x for x large
        too_small.select(features_exp,  // softplus(x) ~= exp(x) for x small
                         features_exp.log1p()));
  }
};
```

**Comment by barondra** [comment]
I am having the same problem here with Convolutional layers with softplus activation function at the end of the network as a regressor model. For prediction with lower values, the converted model turns out nicely. Here is examples of [[predicted]] [label]
```
[[87.694496]] [83]
[[79.957184]] [80]
[[71.28555]] [70]
[[72.062096]] [70]
[[71.2224]] [70]
```
However, for the prediction of higher values, a different converted model cannot handle this problem. See examples:
```
[[inf]] [125]
[[inf]] [124]
[[inf]] [140]
[[inf]] [135]
[[inf]] [138]
```

**Comment by pkgoogle** [comment]
Hi @YaoJiayi, similarly this is also handled better with [AI-Edge-Torch](https://github.com/google-ai-edge/ai-edge-torch).

PT/TFLite Program:
```py
import ai_edge_torch
import torch
from torch import nn
import torch.nn.functional as F


class Model(nn.Module):
    def __init__(self):
        super(Model, self).__init__()

    def forward(self, x):
        return F.softplus(x)


sample_input = (torch.tensor([100., 100.], dtype=torch.float32).reshape(1, 2),)
m = Model()

print('PyTorch model output: ', m(*sample_input))

edge_model = ai_edge_torch.convert(m.eval(), sample_input)

print('Lite mode output: ', edge_model(*sample_input))
```

output:
```sh
python convert.py
2024-06-13 20:15:16.656857: I external/local_xla/xla/tsl/cuda/cudart_stub.cc:32] Could not find cuda drivers on your machine, GPU will not be used.
2024-06-13 20:15:16.660374: I external/local_xla/xla/tsl/cuda/cudart_stub.cc:32] Could not find cuda drivers on your machine, GPU will not be used.
2024-06-13 20:15:16.700695: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-06-13 20:15:17.737244: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
WARNING:root:PJRT is now the default runtime. For more information, see https://github.com/pytorch/xla/blob/master/docs/pjrt.md
WARNING:root:Defaulting to PJRT_DEVICE=CPU
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1718309720.031360 1653087 cpu_client.cc:424] TfrtCpuClient created.
PyTorch model output:  tensor([[100., 100.]])
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
W0000 00:00:1718309721.819365 1653087 tf_tfl_flatbuffer_helpers.cc:392] Ignored output_format.
W0000 00:00:1718309721.819399 1653087 tf_tfl_flatbuffer_helpers.cc:395] Ignored drop_control_dependency.
2024-06-13 20:15:21.820413: I tensorflow/cc/saved_model/reader.cc:83] Reading SavedModel from: /tmp/tmpn6y_1jmz
2024-06-13 20:15:21.820601: I tensorflow/cc/saved_model/reader.cc:52] Reading meta graph with tags { serve }
2024-06-13 20:15:21.820622: I tensorflow/cc/saved_model/reader.cc:147] Reading SavedModel debug info (if present) from: /tmp/tmpn6y_1jmz
2024-06-13 20:15:21.826618: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:388] MLIR V1 optimization pass is not enabled
2024-06-13 20:15:21.826862: I tensorflow/cc/saved_model/loader.cc:236] Restoring SavedModel bundle.
2024-06-13 20:15:21.834846: I tensorflow/cc/saved_model/loader.cc:220] Running initialization op on SavedModel bundle at path: /tmp/tmpn6y_1jmz
2024-06-13 20:15:21.837414: I tensorflow/cc/saved_model/loader.cc:462] SavedModel load for tags { serve }; Status: success: OK. Took 17006 microseconds.
2024-06-13 20:15:21.842995: I tensorflow/compiler/mlir/tensorflow/utils/dump_mlir_util.cc:268] disabling MLIR crash reproducer, set env var `MLIR_CRASH_REPRODUCER_DIRECTORY` to enable.
2024-06-13 20:15:21.864291: I tensorflow/compiler/mlir/lite/flatbuffer_export.cc:3531] Estimated count of arithmetic ops: 2  ops, equivalently 1  MACs
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
Lite mode output:  [[100. 100.]]
I0000 00:00:1718309722.810491 1653087 cpu_client.cc:427] TfrtCpuClient destroyed.
```

Let us know if this resolves your issue.
